doc/ReferenceTables.md

```diff
@@ -0,0 +1,153 @@
+# Reference Tables
+
+This document describes a proposed design and implementation guidelines for
+the `Reference Tables` Vitess feature.
+
+The idea is to have a `reference keyspace` that contains a small number of
+`reference tables`, and replicate these tables to every shard of another
+keyspace, the `destination keyspace`. Any update to the reference tables will be
+propagated to the destination keyspace. The reference tables in the destination
+keyspace can then be used directly, in read-only mode (in `JOIN` queries for
+instance). This provides for much better performance than cross-keyspace joins.
+
+Since the data is replicated to every shard on the destination keyspace, the
+write QPS on the reference keyspace is also applied to every shard on the
+destination keyspace. So the change rate in the reference keyspace cannot be
+very high, and so let's also assume it is not sharded.
+
+Vitess already has all the right components to support this scenario, it's just
+a matter of plumbing it the right way. Let's explore the required changes.
+
+## Replication Setup
+
+We can copy all the data and then setup `Filtered Replication` between the
+reference keyspace and each shard of the destination keyspace. This is really
+just a corner case of the vertical splits Vitess already supports.
+
+Action items:
+
+* First, this setup probably needs to be explicitely mentioned somewhere in the
+  topology, not just as SourceShard objects in the destination keyspace, so
+  Vitess can know about this setup at a higher level. Let's add a `repeated
+  ReferenceKeyspace` field to the Keyspace object. Each `ReferenceKeyspace`
+  object contains the name of the reference keyspace, the list of tables to
+  copy, and the UID of the SourceShard entry (the same UID in all shards). By
+  making this a repeated field, the destination keyspace should be able to
+  support multiple reference keyspaces to copy data from, if necessary.
```

Comment:
Well filtered replication is a pull model the way we have it now, the destination master pulls the data from the source shard(s). A push model would mean an extra process somewhere that pulls the data from the source shard(s) and pushes it to the destination shard(s).

I don't see any reason to have an extra process when the destination master can do it... or did you have anything else in mind?